RevenueCat/Example/PurchasesListener.cs

```diff
@@ -74,99 +161,404 @@ private void SwitchUser()
+    }
+
+    void SetSubscriberAttributes()
+    {
+        var purchases = GetComponent<Purchases>();
+        Int32 unixTimestamp = (int)DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalSeconds;
+
+        var timestampString = $"{unixTimestamp}";
+        purchases.SetAttributes(new Dictionary<string, string>
+        {
+            { "timestamp", timestampString },
+        });
+        purchases.SetEmail($"email_{timestampString}@revenuecat.com");
+        purchases.SetPhoneNumber($"{timestampString}");
+        purchases.SetDisplayName($"displayName_{timestampString}");
+        purchases.SetPushToken($"pushtoken_{timestampString}");
+        purchases.SetAdjustID($"adjustId_{timestampString}");
+        purchases.SetAppsflyerID($"appsflyerId_{timestampString}");
+        purchases.SetFBAnonymousID($"fbAnonymousId_{timestampString}");
+        purchases.SetMparticleID($"mparticleId_{timestampString}");
+        purchases.SetOnesignalID($"onesignalId_{timestampString}");
+        purchases.SetAirshipChannelID($"airshipChannelId_{timestampString}");
+        purchases.SetMediaSource($"mediaSource_{timestampString}");
+        purchases.SetCampaign($"campaign_{timestampString}");
+        purchases.SetAdGroup($"adgroup_{timestampString}");
+        purchases.SetAd($"ad_{timestampString}");
+        purchases.SetKeyword($"keyword_{timestampString}");
+        purchases.SetCreative($"creative_{timestampString}");
+        purchases.CollectDeviceIdentifiers();
+        infoLabel.text = "Subscriber attributes set!";
+    }
+
+    void LogInAsTest()
+    {
+        LogIn("test");
+    }
+
+    void LogInAsRandomId()
+    {
+        Guid appUserID = Guid.NewGuid();
+        LogIn(appUserID.ToString());
+    }
+
+    void LogIn(string appUserID)
+    {
+        var purchases = GetComponent<Purchases>();
+        purchases.LogIn(appUserID, (customerInfo, created, error) =>
         {
-            Debug.Log("getProducts " + products);
             if (error != null)
             {
                 LogError(error);
             }
+            else
+            {
+                infoLabel.text = $"created: {created}\n customerInfo:\n{customerInfo.ToString()}";
+            }
         });
+    }
 
-        purchases.SyncPurchases();
-        purchases.SetFinishTransactions(false);
-        Debug.Log("user ID " + purchases.GetAppUserId());
-        Debug.Log("user is anonymous " + purchases.IsAnonymous());
+    void LogOut()
+    {
+        var purchases = GetComponent<Purchases>();
+        purchases.LogOut((customerInfo, error) =>
+        {
+            if (error != null)
+            {
+                LogError(error);
+            }
+            else
+            {
+                infoLabel.text = customerInfo.ToString();
+            }
+        });
     }
 
-    private void ButtonClicked(Purchases.Package package)
+    void CheckIntroEligibility()

```

Comment:
some of these are a little complex because they call several things in a row. 
I think ideally you'd have a button that gets you the offerings, then you tap and you get the packages, then you tap and you get the discounts or intro eligibility and what not. 
but don't know how to do interesting UI in Unity and it seems like too much time to spend on it for the value